orchestrator/aggregation.py

```
@@ -0,0 +1,1167 @@
+        
+        # Validate weights sum to 1.0
+        weight_sum = sum(weights)
+        tolerance = 1e-6
+        if abs(weight_sum - 1.0) > tolerance:
+            msg = f"Cluster weight validation failed: sum={weight_sum:.6f}"
+            logger.error(msg)
+            if self.abort_on_insufficient:
+                raise WeightValidationError(msg)
+        
+        # Calculate weighted average
+        weighted_avg = sum(s * w for s, w in zip(scores, weights))
```

Ruff (0.14.2)

`zip()` without an explicit `strict=` parameter

Add explicit value for parameter `strict=`

(B905)